### PR TITLE
ipsi feature: ipsiListen() + enviornment setupScript

### PR DESCRIPTION
**Pull request by ashwini1025**
Includes an feature implementation of ipsiListen() and dependency setup Script.

### CODE REVIEW

File: ipsi/src/ipsi.c [hunk 1]
```diff
@@ -287,6 +287,41 @@ int ipsiMethodCall(char *appName,char* methodName)
  **/
 int ipsiListen(){
```
Comment (parthoCarleton) [inline ipsi/src/ipsi.c:289]: Keep an eye for code formatting

File: ipsiSetup/ipsiEnvSetup.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+sudo apt-get install -y build-essential cmake  vim libgtk2.0-dev \
+                        libavcodec-dev libavformat-dev \
+                        libswscale-dev libjpeg-dev libpng-dev libtiff5-dev \
+                        libjasper-dev libopencv-dev checkinstall \
+                        libjpeg-dev libjasper-dev libavcodec-dev \
+                        libavformat-dev libswscale-dev libdc1394-22-dev \
+                        libmp3lame-dev libopencore-amrnb-dev \
+                        libopencore-amrwb-dev libtheora-dev libvorbis-dev \
+                        libxvidcore-dev x264 v4l-utils unzip \
+			libdbus-1-dev libdbus-glib-1-dev lcov xterm
```
Comment (parthoCarleton) [inline ipsiSetup/ipsiEnvSetup.sh:23]: Hope xterm work good for demo

File: ipsiSetup/ipsiEnvSetup.sh [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                        libxvidcore-dev x264 v4l-utils unzip \
+			libdbus-1-dev libdbus-glib-1-dev lcov xterm 
+printf "\n"
+echo "***************************************************************************"
+echo "NOTE: Your machine has now required packages to run Demo + Unit Testing tools"
+echo "You are not required to re-Run this option Again"
+echo "If any packages failed to install then contact the Developers or try re-Running this option"
+printf "\n"
+echo "WARNING: If Demo or Unit Testing Framework fails to compile or Run , you might try to re-Run"
+echo "this option or conatct the Developer"
```
Comment (parthoCarleton) [inline ipsiSetup/ipsiEnvSetup.sh:31]: should be contact

**Review by parthoCarleton — CHANGES_REQUESTED** [review]

File: ipsiSetup/ipsiEnvSetup.sh (hunk 2 above)
Comment (ashwini1025) [reply ipsiSetup/ipsiEnvSetup.sh:23]: Yes.

File: ipsiSetup/ipsiEnvSetup.sh (hunk 3 above)
Comment (ashwini1025) [reply ipsiSetup/ipsiEnvSetup.sh]: Done.

**Review by parthoCarleton — APPROVED** [review]

**Pull request merged**